src/Mvc/Mvc.Core/src/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -143,6 +143,11 @@ public async Task BindModelAsync(ModelBindingContext bindingContext)
 
             if (formatter == null)
             {
+                if (AllowEmptyBody && httpContext.Request.ContentLength == 0)
+                {
+                    return;
```

Comment:
I think you might need to do `bindingContext.Result = ModelBindingResult.Success(model: null);`. See the comment on Line 178.

Reply:
Is there a test I should write for this? I'm not sure how this is visible

Reply:
Hmm, I think an actual case where problematic behavior might be observable if if you have a `[FromBody]` property. But maybe we could just write a unit test that verify `IsModelSet == true`?